student.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(id, age, classroom, name = 'Unknown', parent_permission: true)
```

Comment:
Same here, since the `id` will be generated automatically by calling `super` you don't need to pass it in the constructor function. Kindly remove it from the constructor and also from the `super()`